core/controllers/creator_dashboard.py

```diff
@@ -264,20 +248,20 @@ class NotificationsHandler(base.BaseHandler):
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
+    @acl_decorators.can_access_creator_dashboard
```

Comment:
No, this count shows for me on every page, at the top-right. It should only show when you can access the creator dashboard but please verify carefully that in the logged-out case, no request is made to this handler in the first place. Otherwise the frontend needs to be modified.